modules/squants/src/main/scala/pureconfig/module/squants.scala

```diff
@@ -0,0 +1,179 @@
+  implicit val luminousFluxConfigConvert =
+    ConfigConvert.nonEmptyStringConvert[LuminousFlux](LuminousFlux.apply, _.toString)
+  implicit val luminousIntensityConfigConvert =
+    ConfigConvert.nonEmptyStringConvert[LuminousIntensity](LuminousIntensity.apply, _.toString)
+
+  // radio
+  implicit val irradianceConfigConvert =
+    ConfigConvert.nonEmptyStringConvert[Irradiance](Irradiance.apply, _.toString)
+  implicit val radianceConfigConvert =
+    ConfigConvert.nonEmptyStringConvert[Radiance](Radiance.apply, _.toString)
+  implicit val spectralIntensityConfigConvert =
+    ConfigConvert.nonEmptyStringConvert[SpectralIntensity](SpectralIntensity.apply, _.toString)
+  implicit val spectralIrradianceConfigConvert =
+    ConfigConvert.nonEmptyStringConvert[SpectralIrradiance](SpectralIrradiance.apply, _.toString)
+  implicit val spectralPowerConfigConvert =
+    ConfigConvert.nonEmptyStringConvert[SpectralPower](SpectralPower.apply, _.toString)
+
+  // space
+  implicit val angleConfigConvert =
+    ConfigConvert.nonEmptyStringConvert[Angle](Angle.apply, _.toString)
+  implicit val areaConfigConvert =
+    ConfigConvert.nonEmptyStringConvert[Area](Area.apply, _.toString)
+  implicit val lengthConfigConvert =
+    ConfigConvert.nonEmptyStringConvert[Length](Length.apply, _.toString)
+  implicit val solidAngleConfigConvert =
+    ConfigConvert.nonEmptyStringConvert[SolidAngle](SolidAngle.apply, _.toString)
+  implicit val volumeConfigConvert =
+    ConfigConvert.nonEmptyStringConvert[Volume](Volume.apply, _.toString)
+
+  // thermal
+  implicit val temperatureConfigConvert =
+    ConfigConvert.nonEmptyStringConvert[Temperature](parseTemperature, _.toString)
+  implicit val thermalCapacityConfigConvert =
+    ConfigConvert.nonEmptyStringConvert[ThermalCapacity](ThermalCapacity.apply, _.toString)
+
+  // time
+  implicit val frequencyConfigConvert =
+    ConfigConvert.nonEmptyStringConvert[Frequency](Frequency.apply, _.toString)
+  implicit val timeConfigConvert =
+    ConfigConvert.nonEmptyStringConvert[Time](Time.apply, _.toString)
+
+  // This is temporary until https://github.com/typelevel/squants/pull/183 is released (1.2.0 ???)
+  // Without it, ScalaCheck has a pretty easy time breaking the conversion
+  private[squants] def parseTemperature(s: String): Try[Temperature] = {
+    val regex = "([-+]?[0-9]*\\.?[0-9]+(?:[eE][-+]?[0-9]+)?)*°? *(.{1})".r
+    s match {
+      case regex(value, unit) => unit match {
+        case Fahrenheit.symbol | "f" | "F" => Success(Fahrenheit(value.toDouble))
+        case Celsius.symbol | "c" | "C" => Success(Celsius(value.toDouble))
+        case Kelvin.symbol | "k" | "K" => Success(Kelvin(value.toDouble))
+        case Rankine.symbol | "r" | "R" => Success(Rankine(value.toDouble))
+        case unknownUnit => Failure(new Exception(s"PureConfig library programmer error: The regex '$regex' found unit '$unknownUnit' which was not defined."))
```

Comment:
@cranst0n 
This is me being an old paranoid engineer assuming that somebody somewhere is going to screw this up. 
This case that I'm describing would only be hit if somebody changed the regex but forgot to add a matching case for the unit.
The whole point is that the regex and the cases have diverged.
Since there's not a nice way to make the compiler check for that, I was adding a runtime check to give the pureconfig author a hint how we messed up.
Belt and suspenders I call it.
Sorry I confused you. Feel free to ignore my suggestion.